packages/integration/src/github/DefaultGithubCredentialsProvider.test.ts

```diff
@@ -0,0 +1,347 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ScmIntegrations } from '../ScmIntegrations';
+
+const octokit = {
+  paginate: async (fn: any) => (await fn()).data,
+  apps: {
+    listInstallations: jest.fn(),
+    createInstallationAccessToken: jest.fn(),
+  },
+};
+
+jest.doMock('@octokit/rest', () => {
+  class Octokit {
+    constructor() {
+      return octokit;
+    }
+  }
+  return { Octokit };
+});
+
+import { DefaultGithubCredentialsProvider } from './DefaultGithubCredentialsProvider';
+import { RestEndpointMethodTypes } from '@octokit/rest';
+import { DateTime } from 'luxon';
+import { ConfigReader } from '@backstage/config';
+
+let integrations: ScmIntegrations;
+
+describe('DefaultGithubCredentialsProvider tests', () => {
```

Comment:
I feel that these tests are very deep. Testing whether octokit is called correctly would be part of testing the mux etc. At this level it should be enough that it forwards requests to the correct upstream instance.

Reply:
Cool will do.

Reply:
Done.